contracts/cw20-ics20/src/amount.rs

```diff
@@ -1,30 +1,32 @@
 use crate::error::ContractError;
 use cosmwasm_schema::cw_serde;
-use cosmwasm_std::{Coin, Uint128};
-use cw20::Cw20Coin;
+use cosmwasm_std::{Addr, CanonicalAddr, Coin, StdError, StdResult, Uint128};
+use cw20::Cw20CoinVerified;
 use std::convert::TryInto;
 
 #[cw_serde]
 pub enum Amount {
     Native(Coin),
-    // FIXME? USe Cw20CoinVerified, and validate cw20 addresses
-    Cw20(Cw20Coin),
```

Comment:
I wouldn't do this. You can see the problem with trying to parse a string and suddenly needing `Deps` to do the verification.

I would rather keep Amount as is, and then make an `VerifiedAmount` type, with

```rust
impl Amount {
  pub fn verify(self, deps: Deps) -> StdResult<VerifiedAmount> { /* ... */ }
}
```

Moving parsing and verifying into two steps.